### DISCUSSION TITLE
How to align each process before algo.compute()?

### DISCUSSION THREAD

**Discussion by bobjiang82** [intel/scikit-learn-intelex#591] [Q&A] [answered]
I would like to measure the algo.compute() runtime in distributed computing. So I wonder if there's an API like mpi_barrier() to sync each process after data loading and before algo.compute(). Something like below,
```
read_csv()
mpi_barrier()
algo.compute()
```

**Comment by PetrovKP** [comment, marked as answer]
We use mpi to run our algorithms for python. For synchronization across all group members, you can use the MPI functions. For example:
```python
import daal4py as d4p
d4p.daalinit() # for MPI init of daal4py
from mpi4py import MPI
comm = MPI.COMM_WORLD
data = read_csv()
comm.Barrier()
algo = d4p.algorithm()
algo.compute(data)
...
d4p.daalfini() # for MPI finilize of daal4py
```

**Reply by bobjiang82** [reply]
Thank you.
Works like a charm.
